dove/tests/cmd_dove_init_fail.rs

```diff
@@ -0,0 +1,85 @@
+#![cfg(test)]
+
+use std::path::{Path};
+use std::fs::{remove_dir_all};
+use dove::cli::execute;
+
+/// Fail
+/// $ dove init -d incorrectdialect
+/// $ dove build
+/// project: demoproject_46
+#[test]
+fn incorrect_dialect() {
+    // Path to dove folder
+    let dove_folder = {
+        let mut folder = Path::new(".").canonicalize().unwrap();
+        if folder.to_str().unwrap().find("dove").is_none() {
+            folder.push("dove");
+        }
+        folder
+    };
+    // Project name and path
+    let project_name = "demoproject_46";
+    let mut folder = dove_folder.clone();
+    folder.push(project_name);
+    if folder.exists() {
+        remove_dir_all(&folder).expect(&format!(
+            "[ERROR] Couldn't delete project directory: {}",
+            folder.to_str().unwrap()
+        ));
+    }
+
+    // $ dove new demoproject_46 -d incorrectdialect
+    {
+        let args = &["dove", "init", "-d", "incorrectdialect"];
+        let command_string: String = args.join(" ").to_string();
+        assert!(execute(args, dove_folder.clone()).is_err(), "[ERROR] There must be a mistake here. Invalid dialect\r\n[COMMAND] {}\r\n[DIALECT] incorrectdialect", &command_string)
```

Comment:
nit: hard-coding error strings makes difficult updating them. Not a big issue though